### PR TITLE
Make --dir command optional, assume "." default value

### PR DESCRIPTION
**Pull request by TomasHofman**
Resolves https://github.com/wildfly-extras/prospero/issues/87.

The `install` command remains as it was - `--dir` is mandatory and must be an empty dir.

### CODE REVIEW

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.wildfly.prospero.cli;
```
Comment (spyrkob) [inline prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java:1]: Can you add the license header please?

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java [hunk 2]
```diff
@@ -23,4 +29,24 @@ public AbstractCommand(Console console, ActionFactory actionFactory) {
         this.console = console;
         this.actionFactory = actionFactory;
     }
+
+    static Path determineInstallationDirectory(Optional<Path> directoryOption) {
```
Comment (spyrkob) [inline prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java:33]: do the new methods need to be static?

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java (hunk 1 above)
Comment (TomasHofman) [reply prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java:1]: +1 will update my idea settings finally and add the headers in a new commit.

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java (hunk 2 above)
Comment (TomasHofman) [reply prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java:33]: Not necessarily, I like to mark the methods static if they are not dependent on the instances' inner state and don't manipulate the inner state, to highlight that fact. But it's matter of taste... Let me know if you still prefer non-static :].

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java (hunk 1 above)
Comment (TomasHofman) [reply prospero-cli/src/main/java/org/wildfly/prospero/cli/ExecutionExceptionHandler.java:1]: I created new PR where copyright headers are added in all java files: https://github.com/wildfly-extras/prospero/pull/106

File: prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java (hunk 2 above)
Comment (spyrkob) [reply prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/AbstractCommand.java:33]: Fair enough. I tend to try and avoid static methods when possible, but like you say it's matter of taste.
Happy to leave that in

**Pull request merged**